stages/org.osbuild.rpm-ostree

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+
+import json
+import subprocess
+import sys
+
+import osbuild.ostree
+
+
+STAGE_DESC = "Transforms the tree to an ostree layout"
+STAGE_INFO = """
+Uses `rpm-ostree compose` to transform a "normal" file system tree into
+an OSTree conforming layout (see [1]). Among other things the main steps
+are:
+  - moves /etc to /usr/etc
+  - move /boot to /usr/lib/ostree-boot
+  - potentially moving /var/lib/rpm
+  - re-creates the initramfs (via dracut)
+  - adds altfiles module to NSS
+  - Re-compiles SELinux policy (semodule -nB)
+  - Migrates /usr/etc/{passwd, group} to /usr/lib/
+  - Postprocess SELinux policy
+  - Convert /var to use tmpfiles.d
+  - Prepares symlinks
+    - /usr/local -> /var/usrlocal
+    - /var/lib/alternatives -> /usr/lib/alternatives
+    - /var/lib/vagrant -> /usr/lib/vagrant
+  - copies the rpmdb
+
+The configuration options, `etc_group_members` and `machineid_compat`
+correspond to the Treefile[2] options of rpm-ostree. In brief: The
+groups mentioned in `etc_group_members` will be stored in /etc/groups
+instead of /usr/etc/groups (which is read-only). Therefore all groups
+that human users need to be part of. The `machineid_compat` option
+controls whether or not an empty /etc/machine-id should be created.
+If no machine-id is present, this will trigger `ConditionFirstBoot=`
+for systemd, which then eventually will populate /etc/machine-id,
+among other things (e.g. running systemctl preset-all).
+
+[1] https://ostree.readthedocs.io/en/latest/manual/adapting-existing/
+[2] https://rpm-ostree.readthedocs.io/en/latest/manual/treefile/
+"""
+STAGE_OPTS = """
+"required": ["enabled_services"],
+"properties": {
+  "etc_group_members": {
+    "description": "Array of group names to still keep in /etc/group",
+    "type": "array",
+    "items": { "type": "string" }
+  },
+  "machineid_compat": {
```

Comment:
We have a choice here, either [`rpm-ostree`](https://github.com/coreos/rpm-ostree/blob/1cf0d557ae8059e689b1fed670022727e9842288/src/libpriv/rpmostree-postprocess.c#L396) will write an empty one a it will be removed. Once, in the future, we roll our own code that does what `rpm-ostree compose postprocess` does, we don't have to care anymore but currently we need to tell it something. 

I agree that it should not be a option though. Read the machine id and reset it after `rpm-ostree` has run?

Reply:
Sounds good to me, if this is the intended behaviour in the end.

Reply:
This is changed so we detect an existing file and adjust the setting accordingly.